plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/AbstractMetastoreTableOperations.java

```diff
@@ -189,34 +189,22 @@ protected void commitNewTable(TableMetadata metadata)
     {
         String newMetadataLocation = writeNewMetadata(metadata, version + 1);
 
-        Table table;
-        try {
-            Table.Builder builder = Table.builder()
-                    .setDatabaseName(database)
-                    .setTableName(tableName)
-                    .setOwner(owner)
-                    .setTableType(TableType.EXTERNAL_TABLE.name())
-                    .setDataColumns(toHiveColumns(metadata.schema().columns()))
-                    .withStorage(storage -> storage.setLocation(metadata.location()))
-                    .withStorage(storage -> storage.setStorageFormat(STORAGE_FORMAT))
-                    .setParameter("EXTERNAL", "TRUE")
-                    .setParameter(TABLE_TYPE_PROP, ICEBERG_TABLE_TYPE_VALUE)
-                    .setParameter(METADATA_LOCATION_PROP, newMetadataLocation);
-            String tableComment = metadata.properties().get(TABLE_COMMENT);
-            if (tableComment != null) {
-                builder.setParameter(TABLE_COMMENT, tableComment);
-            }
-            table = builder.build();
-        }
-        catch (RuntimeException e) {
-            try {
-                io().deleteFile(newMetadataLocation);
-            }
-            catch (RuntimeException ex) {
-                e.addSuppressed(ex);
-            }
-            throw e;
+        Table.Builder builder = Table.builder()
+                .setDatabaseName(database)
+                .setTableName(tableName)
+                .setOwner(owner)
+                .setTableType(TableType.EXTERNAL_TABLE.name())
+                .setDataColumns(toHiveColumns(metadata.schema().columns()))
+                .withStorage(storage -> storage.setLocation(metadata.location()))
+                .withStorage(storage -> storage.setStorageFormat(STORAGE_FORMAT))
+                .setParameter("EXTERNAL", "TRUE")
+                .setParameter(TABLE_TYPE_PROP, ICEBERG_TABLE_TYPE_VALUE)
+                .setParameter(METADATA_LOCATION_PROP, newMetadataLocation);
+        String tableComment = metadata.properties().get(TABLE_COMMENT);
+        if (tableComment != null) {
```

Comment:
why is the cleanup not necessary? Can you extend commit message?

Reply:
Good idea, updated